### PR TITLE
Bring SerializationFieldInfo back to BinaryFormatter in corefx

### PR DESCRIPTION
**Pull request by stephentoub**
We initially left SerializationFieldInfo out of corefx as it wasn't necessary given previous goals around BinaryFormatter; but now it is.  I've ported the code almost as-is from desktop, with some minor style tweaks, and a few necessary changes, e.g. using FieldInfo instead of RuntimeFieldInfo, as the latter isn't accessible.  I've not done anything to minimize allocations / improve perf, etc., as that can be done in the future, and the results of all of this gets cached.

Changing this required regenerating a few of the serialization blobs in the tests.  I also added a test for a case we didn't previously cover, a [Serializable] type derived from a non-serializable type.

cc: @ViktorHofer, @morganbr, @danmosemsft

### CODE REVIEW

File: src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/FormatterServices.cs [hunk 1]
```diff
@@ -2,23 +2,24 @@
...
 namespace System.Runtime.Serialization
 {
     public static class FormatterServices
     {
         private static readonly ConcurrentDictionary<MemberHolder, MemberInfo[]> s_memberInfoTable = new ConcurrentDictionary<MemberHolder, MemberInfo[]>();
 
-        private static FieldInfo[] GetSerializableFields(Type type)
+        private static FieldInfo[] InternalGetSerializableMembers(Type type)
         {
+            Debug.Assert(type != null);
```
Comment (danmosemsft) [inline src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/FormatterServices.cs]: Nit: I suppose this had value as a Contract.Assert, but it doesn't have value as a Debug.Assert as the debugger will stop on the next line if it's null.

Comment (stephentoub) [inline src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/FormatterServices.cs]: I still think such Debug.Asserts have value, especially now that Debug.Assert fail fasts again; code could eat the NullReferenceException that emerges from the next line.

Comment (danmosemsft) [inline src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/FormatterServices.cs]: That is true.

Comment (danmosemsft) [inline src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/FormatterServices.cs]: Although, by that logic, we would put them in far more places than we currently do.

Comment (stephentoub) [inline src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/FormatterServices.cs]: > Although, by that logic, we would put them in far more places than we currently do.

Yes, we would ;)

**Comment by danmosemsft** [comment]
@mmitche the debian queues are backed up?

**Pull request merged**